### PR TITLE
[User management] Enforce 1k user rights limit [DPP-833]

### PR DESCRIPTION
**Pull request by pbatko-da**

### CODE REVIEW

File: ledger/ledger-api-test-tool-on-canton/BUILD.bazel [hunk 1]
```diff
@@ -77,42 +77,7 @@ conformance_test(
...
-            # Also exclude "optional tests" - which are run separately below
-            "CompletionDeduplicationInfoITCommandService",
-            "CompletionDeduplicationInfoITCommandSubmissionService",
-            "CommandDeduplicationParallelIT",
-            "TLSOnePointThreeIT",
-            "TLSAtLeastOnePointTwoIT",
-            "MonotonicRecordTimeIT",  # KV-utils specific
-            "ContractIdIT",  # temporarily disabled due to changed error codes
-        ]),
+        "--include=UserManagementServiceIT",
```
Comment (meiersi-da) [inline ledger/ledger-api-test-tool-on-canton/BUILD.bazel:80]: Is this change intentional?

Comment (pbatko-da) [reply ledger/ledger-api-test-tool-on-canton/BUILD.bazel:80]: No, reverting

File: ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala [hunk 2]
```diff
@@ -713,6 +713,30 @@ object LedgerApiErrors extends LedgerApiErrorGroup {
...
+    )
+    object TooManyUserRights
+        extends ErrorCode(
+          id = "TOO_MANY_USER_RIGHTS",
+          ErrorCategory.InvalidGivenCurrentSystemStateOther,
+        ) {
+      case class Reject(_operation: String, userId: String)(implicit
+          loggingContext: ContextualizedErrorLogger
+      ) extends LoggingTransactionErrorImpl(
+            cause = s"cannot ${_operation}, as user \"${userId}\" would have too many rights."
```
Comment (mziolekda) [inline ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala:733]: This formulation assumes that the clients pass the operation text that can be interpreted as a verb or a verb phrase. This can be easily broken inadvertently. I suggest saying 
"${_operation} failed, as user \"${userId}\" would have too many rights."

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/version_service.proto [hunk 3]
```diff
@@ -57,7 +57,9 @@ message FeaturesDescriptor {
 
 }
 
-// Whether the Ledger API server provides the user management service.
 message UserManagementFeature {
+  // Whether the Ledger API server provides the user management service.
   bool supported = 1;
+  // The maximum number of rights that can be assigned to a single user.
```
Comment (mziolekda) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/version_service.proto:63]: Please say what happens when the value is not set or set to zero. I can see that the ledger-api-test-tool assumes no limit in this case.

File: ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala (hunk 2 above)
Comment (pbatko-da) [reply ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala:733]: I'm a bit reluctant to change this as this is the wording we are already using for two other errors (`UserNotFound` and `UserAlreadyExists`). Could do a small separate PR to change all three consistently if you wish.

Comment (meiersi-da) [reply ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala:733]: @pbatko-da : feel free to change it in this PR for all of them. Better are messages are good to have, and givne that we don't change the error-id it is also a change that's always OK to do. Moreover, these two errors have been introduced as part of the same line of work that you are currently working on.

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/version_service.proto (hunk 3 above)
Comment (pbatko-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/version_service.proto:63]: Ok. Added an explicit note about `0` value and changed to an unsigned type.

File: ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala (hunk 2 above)
Comment (pbatko-da) [reply ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala:733]: I noticed that we don't have any tests that would assert what these messages exactly contain.
I'd say testing it in test-tool is inappropriate because the exact message wording is not a public API.
The other option is creating a unit test for `com.daml.platform.apiserver.services.admin.ApiUserManagementService`. 
Wdyt?

Comment (meiersi-da) [reply ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala]: I would not test this as unit test, but instead make sure that you've seen an example message and are OK with it.

At least from where I'm sitting looking at the actual log output and thinking of how to make it better provides more value than enshrining the current messages into a unit test. @mziolekda what do you think?

Comment (mziolekda) [reply ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala]: agree with @meiersi-da

**Review by mziolekda — APPROVED** [review]
